creativecommons/vocabulary
# Create Reusable Navbar and Footer Using JavaScript for Code Consistency and Maintenance

Fixes [Feature] Create Reusable Navbar and Footer Using JavaScript for Code Consistency and Maintenance #74 
 by @Vanessa082 

## Description
This pull request refactors the website structure by creating separate `navbar.html` and `footer.html` files. This reduces code redundancy and simplifies maintenance by dynamically loading these components into each page using JavaScript.

## Technical details
The implementation utilizes the `fetch()` method in JavaScript to load the navbar and footer HTML files, ensuring consistency across all pages without duplicating code. A new `load-header-footer.js` file has been added to the `utils` folder for fetching these components, and the script links in `vocabulary.js` have been updated to the module type.

## Tests
1. Navigate to different pages of the website.
2. Verify that the navbar and footer are consistent across all pages.
3. Ensure that there are no JavaScript errors in the console.

## Screenshots


## Checklist

[best_practices]: https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin




For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
